src/Mail/MailInterface.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace ZfrMail\Mail;
+
+/**
+ * Simple interface for a mail
+ *
+ * @author Michaël Gallego
+ */
+interface MailInterface
+{
+    /**
+     * Add a new "from" address and return a new mail
+     *
+     * @param  string $from
+     * @return MailInterface
+     */
+    public function withFrom(string $from): MailInterface;
+
+    /**
+     * Get the "from" email
+     *
+     * @return string
+     */
+    public function getFrom(): string;
+
+    /**
+     * Add a new "to" address and return a new mail
+     *
+     * @param  string $to
+     * @return MailInterface
+     */
+    public function withTo(string $to): MailInterface;
+
+    /**
+     * Get the "to" email
+     *
+     * @return string
+     */
+    public function getTo(): string;
+
+    /**
+     * Add a new "cc" addresses and return a new mail
+     *
+     * @param  array $cc
+     * @return MailInterface
+     */
+    public function withCc(array $cc): MailInterface;
+
+    /**
+     * Get the "cc" emails
+     *
+     * @return array
+     */
+    public function getCc(): array;
+
+    /**
+     * Add a new "bcc" addresses and return a new mail
+     *
+     * @param  array $bcc
+     * @return MailInterface
+     */
+    public function withBcc(array $bcc): MailInterface;
+
+    /**
+     * Get the "bcc" emails
+     *
+     * @return array
+     */
+    public function getBcc(): array;
+
+    /**
+     * Set new options (the options are related to the mailer you use)
+     *
+     * @param  array $options
+     * @return MailInterface
+     */
+    public function withOptions(array $options): MailInterface;
```

Comment:
Mhmm what kind of options? Maybe it should be configured at mailer construction time?